src/main/kotlin/graphics/scenery/primitives/ParticleGlyphs.kt

```diff
@@ -92,4 +98,21 @@ class ParticleGlyphs @JvmOverloads constructor(var positions: List<Vector3f>, va
 
         return newMaterial
     }
+
+    fun updatePositions(positions: FloatBuffer)
+    {
+        geometry{
+            vertices = positions
```

Comment:
Careful here. In the constructor, you copy the data from the positions object you hand over to vertices, here, you only change the reference. This might come back and bite you.

Reply:
Okay, so I need to get the reference here. thought kotlin is working with references all the time, if not explicitly said to do differently?